MathlibTest/LintDocstring.lean

```diff
@@ -0,0 +1,60 @@
+import Mathlib.Tactic.AdaptationNote
+import Mathlib.Tactic.Linter.DocString
+
+/-! Tests for the `docstring` linter -/
+
+set_option linter.style.docString true
+
+#adaptation_note /--This comment is not inspected by the `docString` linter.-/
+
+example : True := by
+  #adaptation_note /--   This comment is not inspected by the `docString` linter.
+  -/
+  trivial
+
+/--
+warning: error: doc-strings should start with a single space or newline
+note: this linter can be disabled with `set_option linter.style.docString false`
+-/
+#guard_msgs in
+/--Missing space -/
+example : Nat := 1
+
+/--
+warning: error: doc-strings should end with a single space or newline
+note: this linter can be disabled with `set_option linter.style.docString false`
+-/
+#guard_msgs in
+/-- Missing ending space-/
+example : Nat := 1
+
+/--
+warning: error: doc-strings should start with a single space or newline
+note: this linter can be disabled with `set_option linter.style.docString false`
+-/
+#guard_msgs in
+/--  Two starting spaces -/
+example : Nat := 1
+
+/--
+warning: error: doc-strings should end with a single space or newline
+note: this linter can be disabled with `set_option linter.style.docString false`
+-/
+#guard_msgs in
+/--
+Two ending spaces  -/
+example : Nat := 1
+
+/--
+warning: error: doc-strings should end with a single space or newline
+note: this linter can be disabled with `set_option linter.style.docString false`
+-/
+#guard_msgs in
+/-- Let's give an example.
+```lean4
+def foo : Bool := by
+  sorry
+```
+
```

Comment:
Can you also add a test for a string ending with a comma?

Reply:
Test added!